scripts/update_submods_for_cdc_autopush.sh

```diff
@@ -14,35 +14,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Update and merge submodules temporarily, exporting a combination of short
+# revision hashes (website-mixer-import) to a temp file at the specified path.
+# The value in this file can be used to label autopush builds.
 
-# Creates a new custom DC image and tags it latest.
-
-# Usage: From root, ./scripts/build_custom_dc_and_tag_latest.sh
-
-# The latest image = gcr.io/datcom-ci/datacommons-website-compose:latest
+# Usage: From root, ./scripts/update_submods_for_cdc_autopush.sh $IMAGE_LABEL_PATH
 
 set -e
 set -x
 
+# Check for image label, which is set after submodules are updated.
+image_label_path=$1
+if [[ $image_label_path = "" ]]; then
+  echo "Expected positional argument with image label file path."
+  exit 1
+fi
+
+# Get the website short commit hash before it changes due to a temp commit.
+website_rev="$(git rev-parse --short HEAD)"
+
 # Initialize all submodules, then merge them with their masters and update
 # their pinned versions (locally only).
 ./scripts/update_git_submodules.sh
 ./scripts/merge_git_submodules.sh
 
 # Configure Git to create commits with Cloud Build's service account
-git config user.email $(gcloud auth list --filter=status:ACTIVE --format='value(account)')
+git config user.email "$(gcloud auth list --filter=status:ACTIVE --format='value(account)')"
 
 git commit --allow-empty -am "DO NOT PUSH: Temp commit to update pinned submod versions (empty if submods are already up-to-date)"
 
-website_rev="$(git rev-parse --short HEAD)"
 mixer_rev="$(git rev-parse --short HEAD:mixer)"
 import_rev="$(git rev-parse --short HEAD:import)"
 image_label="${website_rev}-${mixer_rev}-${import_rev}"
 
-# Build a new image and push it to Container Registry, tagging it as latest
-docker build -f build/web_compose/Dockerfile \
-          --tag gcr.io/datcom-ci/datacommons-website-compose:${image_label} \
-          --tag gcr.io/datcom-ci/datacommons-website-compose:latest \
-          .
-docker push gcr.io/datcom-ci/datacommons-website-compose:${image_label}
-docker push gcr.io/datcom-ci/datacommons-website-compose:latest
+echo "$image_label" >"$image_label_path"
```

Comment:
Is it possible to have this script only echo the image label the end and have the cloud build yaml write it to a temp file? The other scripts can then just accept the label as an argument with the cloud build yaml again reading from the temp file instead of the scripts.

Reply:
I think this would also require getting rid of `set -x` which prints all commands as the script runs for ease of debugging logs. Is there a reason not to use a file to store this value? It's temporary only for each run of the trigger.

Reply:
Can we redirect just the last echo to a file? `./test.sh | tail -1 > test.txt`

The reason to use standard arguments vs files is to keep the scripts independent of cloud build specifics as much as possible. It arguably keeps the scripts simpler (this is of course subjective and YMMV!) and allows us to run in other contexts easily.

Reply:
Makes sense! Updated, and also updated [planned autopush cloudbuild changes](https://paste.googleplex.com/6181631964741632).

Reply:
In the cloudbuild file, should the path be `workspace/...` or `/workspace/...`?

Reply:
You're right it should be `/workspace/...`! I'll update it before pushing that change. Thanks!!